### PR TITLE
[Frontend] Publish  Prometheus metrics in run_batch API

### PR DESCRIPTION
**Pull request by pooyadavoodi**
The run_batch API currently doesn't publish Prometheus metrics because run_batch is not part of api_server that creates a Prometheus client in the server. This PR runs an http server using prometheus_client so that the metrics that get published by LLMEngine at the endpoint /metrics can be served.

### CODE REVIEW

File: vllm/entrypoints/openai/run_batch.py [hunk 1]
```diff
@@ -59,6 +60,8 @@ def parse_args():
                         'ID numbers being printed in log.'
                         '\n\nDefault: Unlimited')
 
+    parser.add_argument("--port", type=int, default=8000, help="port number")
```
Comment (alugowski) [inline vllm/entrypoints/openai/run_batch.py:63]: Should be optional.

Comment (pooyadavoodi) [reply vllm/entrypoints/openai/run_batch.py]: Done

**Review by simon-mo — APPROVED** [review]
Thanks. I assume this will enable all the metrics from the LLM engine?

**Comment by pooyadavoodi** [comment]
> Thanks. I assume this will enable all the metrics from the LLM engine?

I think it does include all of them. That being said, I had a hard time to verify them in the unit test because the test needs to send metrics request to the http server in a short period of time where the server is online and metrics are published. 

I manually tested that the metrics get published but I might have missed to check every single one. 

I can check this later and look into missing metrics if any, in a separate PR if that's okay?

**Pull request merged**